### PR TITLE
Feature/39521

### PR DESCRIPTION
**Pull request by UrvashiChaudhary**
Define .env file with following properties:

PORT=9000
NODE_ENV=dev

Create a folder src

Create a file Server.ts inside src

Define a class with the following methods and properties:

- config: property, It will be set by constructor from parameter received.
- bootstrap: call different methods needed for bootstrapping of app.
It will call setupRoutes for now.
It will return this.
- setupRoutes: Method to setup routes
We will define a route(of type get) '/health-check' which will return 'I am OK' as response.
- run: Method to run the express app.
Log error in case of error
Log success message in case of success.
It will return this.

Export Server class.

Create a folder config inside src
- Create a file IConfig.ts which contains interface of config (properties like port and env)
- Create a file configuration.ts
Use dotenv module for extracting config from .env file.
Define config using interface(IConfig).
Use Object.freeze for protecting config object against any modifications after the declaration.
Export config.

Create a file index.ts
Import config and Server.
Create an instance of Server by passing config in constructor
Call methods bootstrap and run using instance
App should be running and success message.

### CODE REVIEW

File: src/Server.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// create a class and define methods according to the ticket#39521
+import * as express from "express";
+class Server {
+app
```
Comment (adilsuccessive) [inline src/Server.ts]: app should be of private access modifier.
type of app is missing

File: src/Server.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+return this;
+}
+setupRouts() {
+const { app }=this;
+app.get('/health-check',( req, res, next ) => {
+res.send("I am fine");
+});
+return this;
+}
+run() {
```
Comment (adilsuccessive) [inline src/Server.ts:19]: run method should return this.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+//import the server and config files
+import Server from './Server';
+import config from './config/configuration';
+console.log("config is " , config);
+const server=new Server( { PORT :9000 } );
```
Comment (adilsuccessive) [inline src/index.ts]: pass port from config.

File: src/config/configuration.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { IConfig } from './IConfig';
```
Comment (adilsuccessive) [inline src/config/configuration.ts:4]: IConfig is not used to validate the keys

File: src/config/configuration.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+const envVars = require ('dotenv').config()
```
Comment (adilsuccessive) [inline src/config/configuration.ts]: use import to instead of require.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

File: src/Server.ts (hunk 2 above)
Comment (UrvashiChaudhary) [reply src/Server.ts:19]: done

File: src/index.ts (hunk 3 above)
Comment (UrvashiChaudhary) [reply src/index.ts]: done

File: src/config/configuration.ts (hunk 4 above)
Comment (UrvashiChaudhary) [reply src/config/configuration.ts:4]: resolved

File: src/Server.ts (hunk 2 above)
Comment (adilsuccessive) [reply src/Server.ts:19]: still  the changes does not reflect in the PR

File: src/index.ts (hunk 3 above)
Comment (adilsuccessive) [reply src/index.ts]: still pending.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

File: src/Server.ts (hunk 1 above)
Comment (UrvashiChaudhary) [reply src/Server.ts]: done

File: src/Server.ts (hunk 2 above)
Comment (UrvashiChaudhary) [reply src/Server.ts:19]: done

File: src/index.ts (hunk 3 above)
Comment (UrvashiChaudhary) [reply src/index.ts]: done

File: src/config/configuration.ts (hunk 5 above)
Comment (UrvashiChaudhary) [reply src/config/configuration.ts]: done

File: src/Server.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    return this;
```
Comment (adilsuccessive) [inline src/Server.ts]: Return at the function start terminates the function and aap should not work.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

**Pull request merged**